modules/statistics/php/NDB_Form_statistics.class.inc

```diff
@@ -545,19 +548,48 @@ class NDB_Form_Statistics extends NDB_Form
         } else {
             $Param_Project = '';
         }
-        $subprojects = Utility::getSubprojectsForProject($currentProject);
+        $subprojects[null] = 'All Valid Sub-Projects';
+
+        foreach (Utility::getSubprojectsForProject($currentProject)
+            as $key=>$value) {
+            $subprojects[$key] = $value;
+        }
 
-        $this->tpl_data['Subprojects'] = $subprojects;
         foreach ($subprojects as $key=>$val) {
             $subprojList .= $key.",";
         }
-        $subprojList = substr($subprojList, 0, -1);
-        if (!empty($subprojList)) {
-            $suproject_query ="AND s.SubprojectID IN ($subprojList)";
+
+        $this->tpl_data['Subprojects'] = $subprojects;
+        $currentSubProject = null;
+        $subprojList       = substr($subprojList, 1, -1);
+        if (isset($_REQUEST['BehaviouralSubproject'])
+            && $_REQUEST['BehaviouralSubproject'] != ''
+            && in_array($_REQUEST['BehaviouralSubproject'], array_keys($subprojects))
+        ) {
+            $currentSubProject = $_REQUEST['BehaviouralSubproject'];
+            $this->tpl_data['CurrentSubproject']
+                = array(
+                   'ID'   => $currentSubProject,
+                   'Name' => $subprojects[$currentSubProject],
+                  );
+            $subproject_query     = "AND (s.SubprojectID IS NULL";
+            $subproject_query    .= " OR s.SubprojectID=:spid)";
+            $this->params['spid'] =$currentSubProject;
         } else {
-            $suproject_query ='';
+            $subproject_query ="";
+        }
+        if (!in_array(
```

Comment:
why is this in not on line 569 

i.e. why are you handling the `$_REQUEST['BehaviouralSubproject']` when it might not even be set !! it should be under the check on lines 565-568
```

if (isset($_REQUEST['BehaviouralSubproject'])
  && $_REQUEST['BehaviouralSubproject'] != ''
  && in_array($_REQUEST['BehaviouralSubproject'], array_keys($subprojects))
```